### ISSUE TITLE
Update RestSharp dependency

### ISSUE THREAD

**Issue by timothodge** [smartsheet/smartsheet-csharp-sdk#18]
* https://github.com/smartsheet-platform/smartsheet-csharp-sdk/issues/156
* https://github.com/smartsheet-platform/smartsheet-csharp-sdk/issues/149
* https://github.com/smartsheet-platform/smartsheet-csharp-sdk/issues/143
* https://github.com/smartsheet-platform/smartsheet-csharp-sdk/issues/142

**Comment by timothodge** [comment]
@stejachittineni Been looking into this. There are some major changes if we go from the current RestSharp version 106.15.0 to the next non-preview version 107.0.1  Some of the changes are as follows and here is a link to a reference for some of the changes. https://restsharp.dev/v107/#restsharp-v107

- IRestRequest and IRestResponse have been deprecated for RestRequest and RestResponse. 
- RestClient now takes a RestClientOptions.

I am adding this as notes for what I am trying but also as an FYI.

**Comment by timothodge** [comment]
Here is my work in progress PR https://github.com/smartsheet/smartsheet-csharp-sdk/pull/29

**Comment by stejachittineni** [comment]
@timothodge I will go through this.

**Comment by timothodge** [comment]
In C# Sync there was discussion of maybe we can get rid of the RestSharp dependency and use the built in HttpClient in C# natively.

**Comment by MrKDx** [comment]
When do you plan to release the RestSharp update? We have a version conflict with another library and would love to see the new version as soon as possible.

**Comment by timothodge** [comment]
My PR is now up to date with changes to make everything build. Will need to do some extensive testing and make sure we are ok with them.

**Comment by timothodge** [comment]
Just to keep the details moving. I am also making another PR to bump RestSharp to v109 to leverage a simpleFactory so we can have many RestClients based on baseUrl. Checkout out PR #31

**Comment by timothodge** [comment]
Hey all, 

was able to condense the changes down to about 5 files in #31 . Please take a look and help me debug if you get a chance.

**Comment by MrKDx** [comment]
Hey @timothodge, thank you. We tried the 3.1.0-beta and it seems to work. Only if the APIAccessToken is empty an exception is thrown but with an empty error response. 

System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (There was an issue connecting. Error response: )
  Source=System.Private.CoreLib
  StackTrace:
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Smartsheet.Api.Internal.Http.DefaultHttpClient.Request(HttpRequest smartsheetRequest)
   at Smartsheet.Api.Internal.AbstractResources.ListResourcesWithWrapper[T](String path)
   at Smartsheet.Api.Internal.SheetColumnResourcesImpl.ListColumns(Int64 sheetId, IEnumerable`1 include, PaginationParameters paging, Nullable`1 level)
   at Smartsheet.Api.Internal.SheetColumnResourcesImpl.ListColumns(Int64 sheetId, IEnumerable`1 include, PaginationParameters paging)
   at Backend.Service.Services.SmartSheetService.<>c__DisplayClass5_1.<GetProjectInfofromSmartSheetAsync>b__1() in C:\backend\Backend.Service\Services\SmartSheetService.cs:line 56
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)

  This exception was originally thrown at this call stack:
    Smartsheet.Api.Internal.Http.DefaultHttpClient.RequestAsync(Smartsheet.Api.Internal.Http.HttpRequest)

Inner Exception 1:
HttpClientException: There was an issue connecting. Error response: 


Code: 

new SmartsheetBuilder()
        .SetHttpClient(new RetryHttpClient())
        .SetAccessToken("TOKEN")
        .SetDateTimeFixOptOut(true)
        .Build();

_client.SheetResources.ColumnResources.ListColumns(sheetId, null, new PaginationParameters(true, null, null));

The exception says DefaultClient. Is the SetHttpClient still working?

**Comment by lifebeyondfife** [comment]
Thanks for the response @MrKDx. To repeat back what you're reporting, if the call to SetAccessToken doesn't contain a token, an exception is thrown i.e. `SetAccessToken()`, and that exception doesn't contain any details?

We have more development on this feature as a consequence of upgrading RestSharp, so we appreciate the feedback. I'd like to confirm though, does 3.1.0-beta unblock your previous issue with the SDK for now?

**Comment by MrKDx** [comment]
Yes, it fixed our issue. We have to different libraries (Smartsheet and OpenAPI generated client) using RestSharp in different versions. 

The exception is thrown not at SetAccessToken() but at the first "real" API call (_client.SheetResources.ColumnResources.ListColumns(sheetId, null, new PaginationParameters(true, null, null)))
